src/System.Device.Gpio/System/Device/Gpio/Drivers/LibgpiodDriver.cs

```diff
@@ -0,0 +1,195 @@
+        {
+            if (mode == PinMode.InputPullDown || mode == PinMode.InputPullUp)
+            {
+                throw new PlatformNotSupportedException("This driver is generic so it does not support Input Pull Down or Input Pull Up modes.");
+            }
+            MutableTuple<PinMode, Process> pinModeAndProcess;
+            if (_pinNumberToPinModeAndProcess.TryGetValue(pinNumber, out pinModeAndProcess))
+            {
+                pinModeAndProcess.Value1 = mode;
+            }
+            else
+            {
+                _pinNumberToPinModeAndProcess.Add(pinNumber, new MutableTuple<PinMode, Process>(mode, null));
+            }
+        }
+
+        protected internal override WaitForEventResult WaitForEvent(int pinNumber, PinEventTypes eventType, CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();  
+        }
+
+        protected internal override void Write(int pinNumber, PinValue value)
+        {
+                // --mode=signal would tell the process set the value and wait for SIGINT or SIGTERM which needed when setting HIGH value 
+            string command = value == PinValue.High ? $"--mode=signal {GpioChip0} {pinNumber}=1" : $"{GpioChip0} {pinNumber}=0";
+
+            Process gpioSetCommand = new Process();
+            gpioSetCommand.StartInfo.FileName = GpioSet;
+            gpioSetCommand.StartInfo.Arguments = command;
+
+            MutableTuple<PinMode, Process> pinModeAndProcess;
+            if (_pinNumberToPinModeAndProcess.TryGetValue(pinNumber, out pinModeAndProcess))
+            {
+                Process oldProcess = pinModeAndProcess.Value2;
+                if (oldProcess != null)
+                {
+                    oldProcess.Kill();
+                    oldProcess.WaitForExit(1);
+                }
+                gpioSetCommand.Start(); // we need to start the pocess after killing any existing one for the given pin
+                if (PinValue.Low == value) {
+                    pinModeAndProcess.Value2 = null;
+                }
+                else
+                {
+                    pinModeAndProcess.Value2 = gpioSetCommand;
+                }
+            }
+            else 
+            {
+                gpioSetCommand.Start();
+                if (PinValue.High == value)
+                {
+                    _pinNumberToPinModeAndProcess.Add(pinNumber, new MutableTuple<PinMode, Process>(PinMode.Output, gpioSetCommand));
+                }
+                
+            }
+        }
+
+        private static bool WaitForProcessToExit(Process process, string command, int waitMilliseconds)
+        {
+            process.WaitForExit(waitMilliseconds);
+            if (process.HasExited)
+            {
+                if (process.ExitCode != 0)
+                {
+                    throw new Exception($"Unknown error occured while running command: {command}");
```

Comment:
As a rule please do not throw the base exception type. Pick a reasonable type, in this case probably `InvalidOperationException`

Reply:
BTW @joperezr you might consider starting to put resources in a .resx (strongly typed) like CoreFX. It makes it easier if one day you localize, it also makes it easier to reuse strings (rather than repeating them) and to review all strings.

Reply:
yeah, was not sure what exception to use

Reply:
> you might consider starting to put resources in a .resx (strongly typed) like CoreFX

Yup, that is the plan. I will do a global pass for this in a future PR.